### PR TITLE
Deprecated `RuntimeCacheTrait` and replaced its usage

### LINKED ISSUE #11339: [Improvement]: Use `RuntimeCacheTrait` whenever possible instead of dealing with `Cache\Runtime` directly
### Improvement description

`RuntimeCacheTrait` was introduced by #11254 

There are multiple places in the code where `Pimcore\Cache\Runtime` is used directly in models, this should be changed to the new trait.

### PR DESCRIPTION
**Pull request by aashan10**
## Changes in this pull request  
Resolves #11339

## Additional info  
As per the discussions on the mentioned issue, `RuntimeCacheTrait` has been deprecated and its usage has been removed from the core.

### CODE REVIEW

File: lib/Cache/RuntimeCacheTrait.php [hunk 1]
```diff
@@ -46,6 +52,7 @@ public static function getCacheEnabled(): bool
      *
      * @param mixed $config
      * @param string $cacheKey
+     * @deprecated
```
Comment (dvesh3) [inline lib/Cache/RuntimeCacheTrait.php:65]: we should trigger a deprecation warning in methods like:
```
trigger_deprecation(
    'pimcore/pimcore',
    '10.5.0',
    sprintf('%s is deprecated and will be removed in Pimcore 11. Use \Pimcore\Cache\Runtime::set() instead.', __METHOD__)
);
```

Comment (aashan10) [reply lib/Cache/RuntimeCacheTrait.php:65]: Thanks! Will change this accordingly!

Comment (aashan10) [reply lib/Cache/RuntimeCacheTrait.php:65]: `RuntimeCacheTrait` has been deprecated with deprecation warnings

File: lib/Cache/RuntimeCacheTrait.php [hunk 2]
```diff
@@ -18,26 +18,42 @@
...
     /**
      * @return bool
+     * @deprecated
      */
     public static function getCacheEnabled(): bool
     {
+        trigger_deprecation(
+            'pimcore/pimcore',
+            '10.5.0',
+            sprintf('%s is deprecated and will be removed in Pimcore 11. Use \Pimcore\Cache\Runtime::set() instead.', __METHOD__)
```
Comment (dvesh3) [inline lib/Cache/RuntimeCacheTrait.php]: Please remove "Use \Pimcore\Cache\Runtime::set() instead."

File: lib/Cache/RuntimeCacheTrait.php [hunk 3]
```diff
@@ -18,26 +18,42 @@
...
     /**
      * @param bool $cacheEnabled
+     * @deprecated
      */
     public static function setCacheEnabled(bool $cacheEnabled): void
     {
+        trigger_deprecation(
+            'pimcore/pimcore',
+            '10.5.0',
+            sprintf('%s is deprecated and will be removed in Pimcore 11. Use \Pimcore\Cache\Runtime::set() instead.', __METHOD__)
```
Comment (dvesh3) [inline lib/Cache/RuntimeCacheTrait.php]: Please remove "Use \Pimcore\Cache\Runtime::set() instead."

File: lib/Cache/RuntimeCacheTrait.php [hunk 4]
```diff
@@ -60,16 +82,23 @@ private static function setCache(mixed $config, string $cacheKey): void
...
      *
      * @param string $cacheKey
+     * @deprecated
      */
     private static function removeCache(string $cacheKey): void
     {
+        trigger_deprecation(
+            'pimcore/pimcore',
+            '10.5.0',
+            sprintf('%s is deprecated and will be removed in Pimcore 11. Use \Pimcore\Cache\Runtime::set() and \Pimcore\Cache::remove() instead.', __METHOD__)
```
Comment (dvesh3) [inline lib/Cache/RuntimeCacheTrait.php]:
```suggestion
            sprintf('%s is deprecated and will be removed in Pimcore 11. Use \Pimcore\Cache\Runtime::set() with null value and \Pimcore\Cache::remove() instead.', __METHOD__)
```

File: lib/Cache/RuntimeCacheTrait.php (hunk 2 above)
Comment (aashan10) [reply lib/Cache/RuntimeCacheTrait.php]: Thanks! This is resolved now!

File: lib/Cache/RuntimeCacheTrait.php (hunk 3 above)
Comment (aashan10) [reply lib/Cache/RuntimeCacheTrait.php]: Thanks! This is resolved now!

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@aashan10 one last thing.. please add it to the upgrade notes here https://github.com/pimcore/pimcore/blob/10.x/doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md#1050

**Comment by aashan10** [comment]
@dvesh3 thank you! I've added the changes to the upgrade notes as well!

**Pull request merged**